lib/galaxy/managers/cloud.py

```diff
@@ -221,6 +221,8 @@ def upload(self, trans, history_id, provider, bucket, objects, credentials, inpu
             params = Params(self._get_inputs(obj, key, input_args), sanitize=False)
             incoming = params.__dict__
             history = trans.sa_session.query(trans.app.model.History).get(history_id)
+            if not history:
```

Comment:
Not related to this PR, but I think the docstring on line 173 should not say `(encoded)`, right ?